src/internet_identity/src/main.rs

```diff
@@ -459,13 +468,22 @@ fn authenticated_anchor_operation<R>(
 
 /// Checks if the caller is authenticated against the anchor provided and returns a reference to the device used.
 /// Traps if the caller is not authenticated.
-fn trap_if_not_authenticated(anchor: &Anchor) -> &Device {
+fn trap_if_not_authenticated(anchor_number: AnchorNumber, anchor: &Anchor) -> &Device {
+    state::with_temp_keys_mut(|temp_keys| temp_keys.prune_expired_keys());
```

Comment:
I missed this line originally, since it's quite far away from the `if` below. Would it make sense to have `check_temp_key` do the pruning?